### PR TITLE
DAOS-7352 tests: Fixing the launch.py archiving of doas config files

### PR DESCRIPTION
**Pull request by phender**
Updating launch.py to archive DAOS configuration files from the
/etc/daos directory on each host.

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -1183,10 +1183,9 @@ def archive_config_files(avocado_logs_dir, args):
     print("Archiving config files from {} in {}".format(host_list, destination))
 
     # Copy any config files
-    base_dir = get_build_environment(args)["PREFIX"]
-    configs_dir = get_temporary_directory(args, base_dir)
+    configs_dir = os.path.join(os.sep, "etc", "daos")
```
Comment (brianjmurrell) [inline src/tests/ftest/launch.py:1186]: Hrm.  How does this operate in a _local-install_ environment? AFAIU we have not yet sun-setted that.  I suspect we want to keep it for developers running locally and possibly for code coverage test runs that need a "local-install"-like environment.

Comment (phender) [reply src/tests/ftest/launch.py]: I've updated the changes to gather config files from either location:

- https://github.com/daos-stack/daos/blob/9f7773bab43e82a09e4d9dae75dcea45172ddd6c/src/tests/ftest/launch.py#L967-L975
- https://github.com/daos-stack/daos/blob/9f7773bab43e82a09e4d9dae75dcea45172ddd6c/src/tests/ftest/launch.py#L977-L983

**Review by brianjmurrell — DISMISSED** [review]

**Review by saurabhtandan — DISMISSED** [review]
LGTM

**Comment by phender** [comment]
There was one test failure in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5672/3/:
- Test Hardware / Functional Hardware Large / FTEST_daos_test.DaosCoreTestDfs.1-./daos_test/daos_core_test_dfs.py - opened  https://jira.hpdd.intel.com/browse/DAOS-7626

**Comment by phender** [comment]
There is one thing I just noticed.  Since the https://github.com/daos-stack/daos/blob/fafe07dc65ea848535466d5bda9deac15bdecbb0/src/tests/ftest/launch.py#L977-L983 code collects any `/etc/daos/daos_*.yml` file it also picks up any default config file (from the RPM install) or any previously modified config file from a host that was used in a different role by another tests (e.g. a host running a server in this test that ran an agent in a previous test). For example: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5672/3/artifact/Functional%20Hardware%20Large/daos_test/daos_core_test_dfs.py/daos_configs/

One possible solution would be to reduce the host list and wildcard matching for the `archive_files()` call.  For example, only use the hosts specified in the `test_clients:` section of the test yaml file to archive the `/etc/daos/daos_{agent|control}*.yml` files.  One concern - although I'm not aware of a case today which would be impacted by this - is that using a more restrictive config file name could prevent the capture of a file if it did not follow the expected naming convention.  We do have https://github.com/daos-stack/daos/blob/9f87976898ee9b6a16f312127ace61a6e6898f26/src/tests/ftest/util/general_utils.py#L1127-L1138 and current uses, like https://github.com/daos-stack/daos/blob/master/src/tests/ftest/pool/destroy_tests.py#L386-L389, do pass an argument that fits this naming convention.

@brianjmurrell @saurabhtandan do you feel this would be to restrictive?

**Comment by brianjmurrell** [comment]
> There is one thing I just noticed. Since the
> 
> https://github.com/daos-stack/daos/blob/fafe07dc65ea848535466d5bda9deac15bdecbb0/src/tests/ftest/launch.py#L977-L983
> 
> code collects any `/etc/daos/daos_*.yml` file it also picks up any default config file (from the RPM install) or any previously modified config file from a host that was used in a different role by another tests (e.g. a host running a server in this test that ran an agent in a previous test).

Could that dir not be cleaned of matching files before populating it with new files for a new test run?

Otherwise your proposal seems sufficient.

**Comment by phender** [comment]
> Could that dir not be cleaned of matching files before populating it with new files for a new test run?
> 
> Otherwise your proposal seems sufficient.

We could clean out the `/etc/daos` directory on every host before we run a test script and then collect/archive any files (or just *.yml files) after the execution of the test file.  Th only problem I see with that approach is removing the files after each test would possibly preclude us from being able to run either multiple test scripts or repeatedly calling the same launch.py command using the same servers/agents started once.

**Comment by phender** [comment]
All tests passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-5672/6/

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by brianjmurrell — APPROVED** [review]

**Review by sylviachanoiyee — CHANGES_REQUESTED** [review]
Has this PR been tested with the following pragmas to make sure coverage files are properly archived? Running unit test and 1 test on each func stage (vm and hw) should be sufficient.
Skip-fnbullseye: false
Skip-bullseye: false

**Comment by phender** [comment]
> Has this PR been tested with the following pragmas to make sure coverage files are properly archived? Running unit test and 1 test on each func stage (vm and hw) should be sufficient.
> Skip-fnbullseye: false
> Skip-bullseye: false

Running with bullseye appears to be broken currently - independent of my changes.  There is no way to skip building on CentOS 8 and it fails with bullseye.  In addition it is extremely cumbersome to build with bullseye as it seemingly requires multiple pragmas to be successful.

**Comment by sylviachanoiyee** [comment]
Let's land this and test the coverage archiving separately.

**Comment by brianjmurrell** [comment]
> Running with bullseye appears to be broken currently - independent of my changes. There is no way to skip building on CentOS 8

Could you open a CORCI ticket for that please?

> In addition it is extremely cumbersome to build with bullseye as it seemingly requires multiple pragmas to be successful.

Whatever those extra/multiple pragmas are, they should be implied by the use of some bulls-eye pragma.  Ultimately it sounds like the detection of the bullseye pragma is not being applied to enough stages.  If you could determin what set of pragmas actually do result in a successful build/test, and put those into a ticket, the required changes could be made to eliminate the need for multiple pragmas.

**Comment by phender** [comment]
> Could you open a CORCI ticket for that please?
https://jira.hpdd.intel.com/browse/CORCI-1088

**Pull request merged**